[Bug] Server error when sorting family list

### Bug Description

```
[2026-04-27 11:24:33 +0000] [6625] [ERROR] Error handling request GET /api/families/?locale=de&profile=self&keys=gramps_id,profile,change&page=1&pagesize=24&sort=+surname&rules=%7B%22rules%22%3A%5B%7B%22name%22%3A%22HasRelType%22%2C%22values%22%3A%5B%22Unknown%22%5D%7D%5D%7D
Traceback (most recent call last):
  File "/venv/lib/python3.13/site-packages/gunicorn/workers/sync.py", line 142, in handle
    self.handle_request(listener, req, client, addr)
    ~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.13/site-packages/gunicorn/workers/sync.py", line 185, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/venv/lib/python3.13/site-packages/flask/app.py", line 1536, in __call__
    return self.wsgi_app(environ, start_response)
           ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.13/site-packages/flask/app.py", line 1514, in wsgi_app
    response = self.handle_exception(e)
  File "/venv/lib/python3.13/site-packages/flask/app.py", line 1511, in wsgi_app
    response = self.full_dispatch_request()
  File "/venv/lib/python3.13/site-packages/flask/app.py", line 919, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/venv/lib/python3.13/site-packages/flask/app.py", line 917, in full_dispatch_request
    rv = self.dispatch_request()
  File "/venv/lib/python3.13/site-packages/flask/app.py", line 902, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "/venv/lib/python3.13/site-packages/gramps_webapi/api/auth.py", line 44, in wrapper
    return func(*args, **kwargs)
  File "/venv/lib/python3.13/site-packages/flask/views.py", line 110, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)  # type: ignore[no-any-return]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^
  File "/venv/lib/python3.13/site-packages/gramps_webapi/api/resources/__init__.py", line 51, in dispatch_request
    return current_app.ensure_sync(meth)(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.13/site-packages/flask_smorest/response.py", line 89, in wrapper
    current_app.ensure_sync(func)(*args, **kwargs)
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.13/site-packages/webargs/core.py", line 652, in wrapper
    return func(*args, **kwargs)
  File "/venv/lib/python3.13/site-packages/flask_smorest/arguments.py", line 85, in wrapper
    return current_app.ensure_sync(func)(*f_args, **f_kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.13/site-packages/flask_caching/__init__.py", line 426, in decorated_function
    rv = self._call_fn(f, *args, **kwargs)
  File "/venv/lib/python3.13/site-packages/flask_caching/__init__.py", line 183, in _call_fn
    return ensure_sync(fn)(*args, **kwargs)
           ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.13/site-packages/gramps_webapi/api/resources/base.py", line 621, in get
    objects = self.sort_objects(objects, args["sort"], locale=locale)
  File "/venv/lib/python3.13/site-packages/gramps_webapi/api/resources/base.py", line 119, in sort_objects
    return sort_objects(
        self.db_handle, self.gramps_class_name, objects, args, locale=locale
    )
  File "/venv/lib/python3.13/site-packages/gramps_webapi/api/resources/sort.py", line 299, in sort_objects
    objects.sort(key=lookup[sort_key], reverse=reverse)
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.13/site-packages/gramps_webapi/api/resources/sort.py", line 117, in by_family_surname_key
    name = person.get_primary_name()
           ^^^^^^
UnboundLocalError: cannot access local variable 'person' where it is not associated with a value
```

### Gramps Web Version Information

Gramps 6.0.8
Gramps Web API 3.12.1
Gramps Web Frontend 26.4.3
Gramps QL 0.4.0
Sifts 1.3.1
locale: en
multi-tree: true
task queue: true
OCR: true
chat: true


### Verification

- [x] I have searched for existing issues before creating this one

--- a/gramps_webapi/api/resources/sort.py
+++ b/gramps_webapi/api/resources/sort.py
@@ -113,7 +113,7 @@ class Sort:
         person = None
         if obj.father_handle is not None:
             person = self.database.get_person_from_handle(obj.father_handle)
-        elif obj.mother_handle is not None:
+        if person is None and obj.mother_handle is not None:
             person = self.database.get_person_from_handle(obj.mother_handle)
         if person is None:
             return self.locale.sort_key("")
@@ -128,7 +128,7 @@ class Sort:
         person = None
         if obj.father_handle is not None:
             person = self.database.get_person_from_handle(obj.father_handle)
-        elif obj.mother_handle is not None:
+        if person is None and obj.mother_handle is not None:
             person = self.database.get_person_from_handle(obj.mother_handle)
         if person is None:
             return ""

--- a/gramps_webapi/api/resources/sort.py
+++ b/gramps_webapi/api/resources/sort.py
@@ -110,10 +110,13 @@ class Sort:
 
     def by_family_surname_key(self, obj: GrampsObject):
         """Compare by family surname, if equal uses given and suffix."""
+        person = None
         if obj.father_handle is not None:
             person = self.database.get_person_from_handle(obj.father_handle)
         elif obj.mother_handle is not None:
             person = self.database.get_person_from_handle(obj.mother_handle)
+        if person is None:
+            return self.locale.sort_key("")
         name = person.get_primary_name()
         fsn = name.get_surname()
         ffn = name.get_first_name()
@@ -122,10 +125,13 @@ class Sort:
 
     def by_family_soundex_key(self, obj: GrampsObject):
         """Compare by family soundex."""
+        person = None
         if obj.father_handle is not None:
             person = self.database.get_person_from_handle(obj.father_handle)
         elif obj.mother_handle is not None:
             person = self.database.get_person_from_handle(obj.mother_handle)
+        if person is None:
+            return ""
         return soundex(person.get_primary_name().get_surname())
 
     def by_family_type_key(self, obj: GrampsObject):

--- a/gramps_webapi/api/resources/util.py
+++ b/gramps_webapi/api/resources/util.py
@@ -1056,10 +1056,13 @@ def get_soundex(
 ) -> str:
     """Return soundex code."""
     if gramps_class_name == "Family":
+        person = None
         if obj.father_handle is not None:
             person = db_handle.get_person_from_handle(obj.father_handle)
-        elif obj.mother_handle is not None:
+        if person is None and obj.mother_handle is not None:
             person = db_handle.get_person_from_handle(obj.mother_handle)
+        if person is None:
+            return ""
     else:
         person = obj
     return soundex(person.get_primary_name().get_surname())

